include/boost/multiprecision/detail/functions/pow.hpp

```diff
@@ -542,23 +540,17 @@ typename std::enable_if<should_use_log_agm<T>::value>::type eval_log(T& result_x
     // say 3 or 4.
```

Comment:
I switched to comparing the exponents of the difference a_k - b_k and the target tolerance.